### ISSUE TITLE
[Core] Could there be a way of getting the combo to hold burst?

### ISSUE THREAD

**Issue by neongrey** [PunishXIV/WrathCombo#99]
I kind of hesitated to put this forward for Sloth because this is a little more dynamic than before, but I think it might be a better fit for Wrath.

In short, there are times when you might want to hold burst in a fight. Upcoming phase transitions, bosses going untargetable, that sort of thing. Sometimes it will not be optimal to do this, like holding burst for the M4S phase transition, but some groups like to do it anyway. When faced with the choice of bursting on time vs bursting under party buffs, it often makes sense to just wait however long the party wants to wait at the expense of overcapping a resource, etc. (Wildfire is the obvious case, because of course it is). Doing so right now is, in my experience, awkward but doable and requires some finesse midcombat.

It would be cool to have some way to just switch to a fallback combo to hold burst with. There may be other possible applications but idk right now. Anyway, maybe a notion someone would be interested in. Thanks!

**Comment by zbee** [comment]
My proposed solution would be to label "burst" actions, and have a key or command or something to delay all burst commands while held or for X time after activating or until re-pressing  the key or something.

If I were to do it, it'd be after I implement moving of options into each individual class, which would be around January.

**Comment by Whitemoose78** [comment]
Not a dev, but I tinker and have some thoughts.
A solution I quite like (but isn't easily doable for all jobs, and a bit less braindead) that's already in, is the current iteration of paladin having most of it's burst skills tied to "Prefers to use whilst under the effect of Fight or Flight" in the advanced rotation section, but not having FoF ticked.  As long as I don't press FoF myself, it'll just do the basic 123/atonement/holy spirit loop and it'll hold burst indefinitely, effectively.
Similarly, Dancer functions the same but with tech step as a "burst trigger", tho you'd also have the manually trigger standard steps every 30 seconds.
Obviously this solution wouldn't work for all jobs, and it's not so much a one-button combo at that point, as you're timing burst manually for the entire encounter, but I'd say it's a possible approach for some jobs, like Dragoon (trigger being lance charge/geir) that doesn't necessitate a full new function

**Comment by edewen** [comment]
It would likely be different for each class. Bard is a tough one bc the burst phase is tied to wanderers minuet and specific timing of delayed weaves inside that songs start and armys end. So any ammount of holding burst breaks bard as a class. let alone what it would do to wrath.

**Comment by Giblett** [comment]
A possible way to do this would be to allow two sets of settings with a bindable key to swap between sets, so there could be a player set of the roation without 2 minutes starters like bloodfest/barrel stabilizer/etc included if applicable for the class, keeping it user cusotmizable. This would essentially allow you to have a preset that lets you hold burst.

**Comment by zbee** [comment]
If we were to add profiles, we would not relegate them to this - we would want them to be high-end content vs lower end content, or auto-rotation vs manual use, etc.
But thank you for your idea.

**Comment by Giblett** [comment]
> If we were to add profiles, we would not relegate them to this - we would want them to be high-end content vs lower end content, or auto-rotation vs manual use, etc. But thank you for your idea, and for bumping this issue.

Yeah, the idea was not specificaly JUST for this issue, but profiles in general would be nice to have.

**Comment by Battler624** [comment]
Here is how I envision it would work.

* All actions would be put in sections, those sections would be buttons in server bar or enabled/disabled somehow you guys see fit.
* For an action to be usable, it would be required for the section its in to be enabled.
* Sections could be Major CDs (2 Minutes), Minor CDs (1 Minutes), Defensives, Movement, Interrupt.
* A default list of sections would be included (the list above) with hopefully custom sections if required (Hopefully with an IPC to be enabled/disabled via other plugins such as BossMod?)

For example if Playing redmage it would by default use everything but if Major CDs would be disabled, it would prevent the usage of manafication or embolden until "Major CDs" is re-enabled but it would still allow other CDs such as acceleration to be used. And if movement is enabled it would use use corps-a-corps (Dash in + damage) and displacement (damage + Dash out) but if movement is disabled it would only use engagement (damage without dashout)

This would unfortunately make the serverbar bloated but i guess its fine? if not you could make it commands or onscreen buttons (like how QolBar does)

**Comment by zbee** [comment]
As mentioned in the status update above:
> * [Add "Request action to be blacklisted" IPC #850](https://github.com/PunishXIV/WrathCombo/pull/850) will add the ability to hold actions/presets via other plugins (think Splatoon)
>
> * when that gets added, tested, and then built out more, we want to lead that into a full burst holding solution within Wrath

#850 has now been merged, and is available for testing in [1.0.2.23](https://github.com/PunishXIV/WrathCombo/releases/tag/1.0.2.23).
You can visit [this page](https://github.com/PunishXIV/WrathCombo/releases/tag/1.0.2.23) for a list of changes it has *on top* of the changes in [.22](https://github.com/PunishXIV/WrathCombo/releases/tag/1.0.2.22), and details on how to use it.

We'd very much appreciate anyone who could help in testing this release, information on what changes there are to test are included the release notes linked here.

**Comment by qnesp1987** [comment]
I think this is addressed already in #1145 which closes this?

**Comment by zbee** [comment]
@Taurenkey unassign this
